logging/logging.go

```diff
@@ -264,12 +267,28 @@ type loggerRetryer struct {
 	defaultRetryer gax.Retryer
 }
 
+func newLoggerRetryer() gax.Retryer {
+	// Copied from CallOptions.WriteLogEntries in apiv2/logging_client.go.
+	d := gax.OnCodes([]codes.Code{
+		codes.DeadlineExceeded,
+		codes.Internal,
+		codes.Unavailable,
+	}, gax.Backoff{
+		Initial:    100 * time.Millisecond,
+		Max:        60000 * time.Millisecond,
+		Multiplier: 1.30,
+	})
+
+	r := &loggerRetryer{defaultRetryer: d}
+	return r
+}
+
 func (r *loggerRetryer) Retry(err error) (pause time.Duration, shouldRetry bool) {
 	s, ok := status.FromError(err)
 	if !ok {
 		return r.defaultRetryer.Retry(err)
 	}
-	if s.Code() == codes.Internal && strings.Contains(s.Message(), "string field contains invalid UTF-8") {
+	if s.Code() == codes.Internal && strings.Contains(s.Message(), utfErrorString) {
```

Comment:
nit: the internal check is not needed here and/or codes above can be reduced to match.

Reply:
I removed the `codes.Internal` check. Good point.